browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -1487,6 +1487,21 @@ L.CanvasTileLayer = L.Layer.extend({
 				var tooltipInfo = JSON.parse(textMsg.substring(textMsg.indexOf('{')));
 				this._map.uiManager.showDocumentTooltip(tooltipInfo);
 			}
+			else if (tooltipInfo.type === 'autofillpreviewtooltip') {
+
+				var strTwips = textMsg.match(/\d+/g);
+				if (strTwips != null && this._map.isEditMode()) {
+
+					var cellRange = this._map._docLayer._parseCellRange(JSON.stringify(tooltipInfo.celladdress));
```

Comment:
All of these calculations don't depend on CanvasTileLayer data.
Can we do them inside Popup code? so here we can just do `_map.fire(....)` and rest will be done inside popup.

Reply:
All moved to `AutoFillPreviewTooltip.ts`

Done. Thanks!